app/controllers/journals_controller.rb

```diff
@@ -91,14 +93,23 @@ def find_journal
     render_404
   end
 
+  def ensure_permitted
+    permission =
+      case @journal.journable_type
+      when 'WorkPackage' then :view_work_packages
+      when 'Project' then :view_project
+      end
+    do_authorize(permission)
```

Comment:
Most (did not check all) journable classes have a `visible?` method. So it would be an option to use 

```
deny_access unless @journal.journable.visible?
```

This has the benefit of avoiding the list of permissions.

Reply:
On the other hand, if it is kept like this, then I'd think the [check for the `journable_type`](https://github.com/opf/openproject/pull/11925/files#diff-341df1934319c300106bd8dc743cbb8c820b1731a1d4e67783ea170fd808011fR112) could be removed all-together. Or the checks are combined.

Reply:
Thanks, I kept the code like this (without `visible?`) and removed the check for the `journable_type` (and a few other things along it)

Reply:
in 0e634e24d911a36df0407eba52d0fcc0be3b1146